### PR TITLE
update eslint, use babel-eslint, eslint-plugin-react, fixes

### PR DESCRIPTION
**Pull request by hzoo**
While working on https://github.com/babel/babel-eslint/issues/108, @wincent mentioned I could check out react-native for examples of flow type usage.

In order to test I updated eslint": "0.21.2", fix deprecated rules, add config to support es6/jsx/react, add babel-eslint. I figured I could do a PR with some fixes (mostly unused parameters, quotes, semicolons etc) to see if it's wanted?

You should be able to then use `"lint": "./node_modules/.bin/eslint Examples Libraries"` (I didn't fix anything in Libraries yet)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -129,7 +146,7 @@
     "no-undef": 2,                   // disallow use of undeclared variables unless mentioned in a /*global */ block
     "no-undefined": 0,               // disallow use of undefined variable (off by default)
     "no-undef-init": 1,              // disallow use of undefined when initializing variables
-    "no-unused-vars": [1, {"vars": "all", "args": "none"}], // disallow declaration of variables that are not used in the code
+    // "no-unused-vars": [1, {"vars": "all", "args": "none"}], // disallow declaration of variables that are not used in the code
```
Comment (jaredly) [inline .eslintrc]: why was this removed?

Comment (hzoo) [inline .eslintrc]: I should of made a comment for that sorry.

It's replaced with `"react/jsx-uses-vars": 1,` in line 54 because of issues with jsx/eslint. ESLint isn't supporting it so it's recommended to use `eslint-plugin-react`

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md

https://github.com/eslint/eslint/search?q=no-unused-vars+for+jsx&type=Issues&utf8=%E2%9C%93

Comment (jaredly) [inline .eslintrc]: Good to know! However, it looks like this still shouldn't be disabled. The jsx-uses-vars docs say:

```
This rule has no effect if the no-unused-vars rule is not enabled.
```

Comment (hzoo) [inline .eslintrc]: Ah ok - good to know xd.

**Comment by hzoo** [comment]
And from Travis - not sure what can be done about this?

```bash
/Users/travis/build/facebook/react-native/node_modules/jstransform/node_modules/commoner/test/source/widget/share.js: WidgetShare
Duplicate module provider
/Users/travis/build/facebook/react-native/node_modules/babel-eslint/node_modules/babel-core/node_modules/regenerator/node_modules/commoner/test/source/widget/share.js:
```

**Comment by hzoo** [comment]
@jaredly Oops I noticed (you probably did as well?) there's already #259 which does the same as this other than not including fixes for any lint issues.

**Comment by jaredly** [comment]
Yup :) I've integrated the .eslintrc & package.json parts of it into a commit internally (in conjunction with https://github.com/facebook/react-native/pull/259). This will be synced up here soon.

I'm going to hold off on the lint fixes for now -- the extra arguments to functions are good for documentation (how will this function be called) even if they aren't used, and I think the convention is to double-quote jsx params

Thanks for the work!

**Comment by hzoo** [comment]
Ah ok!
For the quotes I guess that's what `jsx-quotes` enforces correctly.
And I guess the extra args shouldn't be warning since `"no-unused-vars": [1, {"vars": "all", "args": "none"}]` shouldn't check for arguments.

**Comment by ccheever** [comment]
the packager now uses babel and the linter uses babel-eslint. is there anything else that needs to be done with this?

**Comment by hzoo** [comment]
No this just fixed a few of the lint errors but that can be done seperately - will close.
